src/smt/smt_engine.cpp

```diff
@@ -3337,6 +3349,31 @@ void SmtEnginePrivate::processAssertions() {
   {
     d_passes["bv-to-bool"]->apply(&d_assertions);
   }
+  if (options::solveBVAsInt() > 0)
+  {
+    if (options::incrementalSolving())
+    {
+      throw ModalException(
+          "solving bitvectors as integers is currently not supported "
+          "incrementally.");
+    }
+    else if (options::solveBVAsInt() > 8)
+    {
+      /**
+       * The granularity sets the size of the ITE in each element
+       * of the sum that is generated for bitwise operators.
+       * For granularity k, the size of the ITE is 2^{2*k}.
+       * Since we don't want to introduce ITEs with unbounded size,
+       * we bound the granularity.
+       */
+      throw ModalException("solve-bv-as-int accepts values from 0 to 8.");
```

Comment:
Is 8 the maximum size we can handle?

Reply:
Note: This does not have anything to do with bit-vector size. It's the partition size in which the bit-vector is partitioned.

Reply:
@mpreiner : correct.
@aniemetz : We haven't tried higher than 8, but this already gives ITEs of size `2^16`.